ee/server/service/devices.go

```diff
@@ -16,6 +18,43 @@ func (svc *Service) RequestEncryptionKeyRotation(ctx context.Context, hostID uin
 	return svc.ds.SetDiskEncryptionResetStatus(ctx, hostID, true)
 }
 
+type migrateMDMDevicePayload struct {
+	Timestamp time.Time `json:"timestamp"`
+	Host      struct {
+		ID             uint   `json:"id"`
+		UUID           string `json:"uuid"`
+		HardwareSerial string `json:"hardware_serial"`
+	} `json:"host"`
+}
+
+func (svc *Service) TriggerMigrateMDMDevice(ctx context.Context, host *fleet.Host) error {
+	ac, err := svc.ds.AppConfig(ctx)
+	if err != nil {
+		return err
+	}
+	if !ac.MDM.EnabledAndConfigured {
+		return fleet.NewMDMNotConfiguredError()
+	}
+	if !ac.MDM.MacOSMigration.Enable {
+		return fleet.NewBadGatewayError("macos_migration", ctxerr.New(ctx, "not enabled"))
+	}
+	if ac.MDM.MacOSMigration.WebhookURL == "" {
+		return fleet.NewBadGatewayError("macos_migration", ctxerr.New(ctx, "no webhook URL configured"))
+	}
+
+	p := migrateMDMDevicePayload{}
+	p.Timestamp = time.Now().UTC()
+	p.Host.ID = host.ID
+	p.Host.UUID = host.UUID
+	p.Host.HardwareSerial = host.HardwareSerial
+
```

Comment:
Are there any other device-specific validations we should perform before making the webhook request?

Reply:
oh yeah, completely forgot about this! I don't think we should send the webhook to any device, first thing that comes to mind is to check for the same conditions as we do here:

https://github.com/fleetdm/fleet/blob/5fb37bff17092b035827cb06d87b29e8a174c530/ee/server/service/devices.go#L43-L49

then, ideally we shouldn't send the same webhook twice, at least not in quick succession. This can be done in a follow up, potentially after discussing in Slack what are our options! maybe I'm overreacting.

Reply:
Not that I can think of.